### PR TITLE
[processor/k8sattributesprocessor] add debug logs

### PR DESCRIPTION
**Pull request by povilasv**
**Description:** <Describe what has changed.>

Adding debug logs to k8sattributesprocessor - at the moment it is impossible to debug when the k8sattributeprocessor fails to add metadata. Adding debug logs for pod identification allows us to see the issue with pod association.
 
**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

Tested locally:

Here are some examples of messages:

```
2022-06-10T07:08:12.818Z        debug   k8sattributesprocessor/processor.go:111 Evaluating Pod Identifier       {"kind": "processor", "name": "k8sattributes", "pipeline": "traces", "pod identifier key": "k8s.pod.ip", "pod identifier value": "10.244.0.179"}
```

```
2022-06-10T07:08:06.610Z        debug   k8sattributesprocessor/processor.go:111 Evaluating Pod Identifier       {"kind": "processor", "name": "k8sattributes", "pipeline": "metrics", "pod identifier key": "", "pod identifier value": ""}  
```
```
2022-06-10T07:08:12.818Z        debug   k8sattributesprocessor/processor.go:123 Getting the pod {"kind": "processor", "name": "k8sattributes", "pipeline": "traces", "pod identifier value": "k8s.pod.ip", "pod": {"Name":"jaeger-http-app-7f7c9657cd-h8rfb","Address":"10.244.0.179","PodUID":"293411d1-5132-46d0-aa6a-df138ca937aa","Attributes":{"":"jaeger-http-app","k8s.deployment.name":"jaeger-http-app","k8s.na
mespace.name":"sys-mon"},"StartTime":"2022-06-09T11:23:02Z","Ignore":false,"Namespace":"sys-mon","Containers":null,"DeletedAt":"0001-01-
01T00:00:00Z"}}       
```
**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/k8sattributesprocessor/processor.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func (kp *kubernetesprocessor) processLogs(ctx context.Context, ld plog.Logs) (p
 // processResource adds Pod metadata tags to resource based on pod association configuration
 func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pcommon.Resource) {
 	podIdentifierKey, podIdentifierValue := extractPodID(ctx, resource.Attributes(), kp.podAssociations)
+	kp.logger.Debug("Evaluating Pod Identifier", zap.String("pod identifier key", podIdentifierKey), zap.Any("pod identifier value", podIdentifierValue))
```
Comment (pmm-sumo) [inline processor/k8sattributesprocessor/processor.go:111]: A nit, we are not typically using capital letters when emitting logs ([example](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/internal/kube/client.go#L143))
Also, "pod identifier key"/"pod identifier value" seem a little bit redundant (considering the context maybe just "key"/"value" are enough) but it's subjective, so feel free to leave it as is or change

File: processor/k8sattributesprocessor/processor.go [hunk 2]
```diff
@@ -118,6 +119,8 @@ func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pco
 
 	if podIdentifierKey != "" {
 		if pod, ok := kp.kc.GetPod(podIdentifierValue); ok {
+
+			kp.logger.Debug("Getting the pod", zap.String("pod identifier value", podIdentifierKey), zap.Any("pod", pod))
```
Comment (pmm-sumo) [inline processor/k8sattributesprocessor/processor.go:123]: same like above, we tend to not start the log message with capital letter

**Comment by pmm-sumo** [comment]
Thanks for the contribution @povilasv I think it's gonna be very useful. There are some nits I called out. Also, we miss changelog entry

**Comment by povilasv** [comment]
@pmm-sumo thanks for taking a look! I've updated PR :dancing_men:

**Review by pmm-sumo — APPROVED** [review]
LGTM. Can you take a second look @dmitryax ?

**Comment by mdrakiburrahman** [comment]
Is there a reason this PR was closed without merging?

**Comment by povilasv** [comment]
uups I incidentally deleted my fork, thats probably closed the PR

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@povilasv please resolve conflicts. I cannot do that because your fork of the repo is done :)

**Comment by povilasv** [comment]
Github doesn't let me connect this PR back to my new fork, so I've create a new PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/11060

Sorry for the hassle

**Comment by pmm-sumo** [comment]
Closing this PR as the code change is now tracked in #11060

**Pull request closed without merging**